src/mqtt.client.ts

```diff
@@ -123,22 +121,39 @@ export class MqttClient<
             );
     }
 
-    public async connect(options?: Resolvable<RegisterClientOptions>): Promise<any> {
+    private async _connect(options?: Resolvable<RegisterClientOptions>): Promise<any> {
         this.expectCreated();
-        this.mqttDebug('Connecting...');
+        this.mqttDebug(`Connecting using transport "${this.transport.constructor.name}"`);
         this.connectResolver = options;
         this.setConnecting();
-
-        await this.transport.connect();
+        try {
+            await this.transport.connect();
+        } catch (e) {
+            this.mqttDebug(`Transport connect error ("${this.transport.constructor.name}")`, e.message);
+            if (this.shouldReconnect()) {
+                return this.setDisconnected(e);
```

Comment:
Maybe add a comment here on why this is done. As I understand, it will return the new promise from calling `connect()` again, but it's confusing seeing the same call twice.

Reply:
in the first case we explicitly return `this.setDisconnected(e)`, it is done because `setDisconnected` calls `reconnect` that calls `connect`, so we return `connect` result anyway.
In the second case we just call `setDisconnected(e)`  and throw an error. 
I was thinking about caching setDisconnected result in variable but it would not be that expressive, and it's one extra line of code.
Not sure what comment to add in code in this case, at my glance it's self-descripted. What do you think to add?

Reply:
> Not sure what comment to add in code in this case, at my glance it's self-descripted. What do you think to add?

Something like 
```
// as the client should reconnect afterwards, this method will call `reconnect` which calls `connect`
```